src/Estimators/src/GyroscopeMeasurementDynamics.cpp

```diff
@@ -0,0 +1,217 @@
+        return false;
+    }
+
+    if (stateVariableHandler.getNumberOfVariables() == 0)
+    {
+        log()->error("{} The state variable handler is empty.", errorPrefix);
+        return false;
+    }
+
+    m_stateVariableHandler = stateVariableHandler;
+
+    if (!checkStateVariableHandler())
+    {
+        log()->error("{} The state variable handler is not valid.", errorPrefix);
+        return false;
+    }
+
+    // Search and save all the submodels containing the sensor
+    for (int submodelIndex = 0; submodelIndex < m_nrOfSubDynamics; submodelIndex++)
+    {
+       if (m_subModelList[submodelIndex].hasGyroscope(m_name))
+       {
+            m_subModelWithGyro.push_back(submodelIndex);
+       }
+    }
+
+    m_covariances.resize(m_covSingleVar.size() * m_subModelWithGyro.size());
+    for (int index = 0; index < m_subModelWithGyro.size(); index++)
+    {
+        m_covariances.segment(index*m_covSingleVar.size(), m_covSingleVar.size()) = m_covSingleVar;
+    }
+
+    m_size = m_covariances.size();
+
+    m_jointVelocityFullModel.resize(m_stateVariableHandler.getVariable("ds").size);
+    m_jointVelocityFullModel.setZero();
+
+    m_subModelJointVel.resize(m_nrOfSubDynamics);
+
+    for (int idx = 0; idx < m_nrOfSubDynamics; idx++)
+    {
+        m_subModelJointVel[idx].resize(m_subModelList[idx].getJointMapping().size());
+        m_subModelJointVel[idx].setZero();
+    }
+
+    m_bias.resize(m_covSingleVar.size());
+    m_bias.setZero();
+
+    m_updatedVariable.resize(m_size);
+    m_updatedVariable.setZero();
+
+    return true;
+}
+
+bool RDE::GyroscopeMeasurementDynamics::setSubModels(const std::vector<SubModel>& subModelList, const std::vector<std::shared_ptr<SubModelKinDynWrapper>>& kinDynWrapperList)
+{
+    constexpr auto errorPrefix = "[GyroscopeMeasurementDynamics::setSubModels]";
+
+    m_subModelList = subModelList;
+    m_subModelKinDynList = kinDynWrapperList;
+
+    if (m_subModelList.size() == 0 || m_subModelKinDynList.size() == 0 || m_subModelList.size() != m_subModelKinDynList.size())
+    {
+        log()->error("{} Wrong size of input parameters", errorPrefix);
+        return false;
+    }
+
+    m_nrOfSubDynamics = m_subModelList.size();
+
+    m_isSubModelListSet = true;
+
+    return true;
+}
+
+bool RDE::GyroscopeMeasurementDynamics::checkStateVariableHandler()
+{
+    constexpr auto errorPrefix = "[GyroscopeMeasurementDynamics::checkStateVariableHandler]";
+
+    if (!m_stateVariableHandler.getVariable("ds").isValid())
+    {
+        log()->error("{} The variable handler does not contain the expected state with name `ds`.", errorPrefix);
+        return false;
+    }
+
+    if (m_useBias)
+    {
+        if (!m_stateVariableHandler.getVariable(m_biasVariableName).isValid())
+        {
+            log()->error("{} The variable handler does not contain the expected state with name `{}`.", errorPrefix, m_biasVariableName);
+            return false;
+        }
+    }
```

Comment:
Here the reason why of this suggestion https://dev.to/zachsnoek/clean-up-your-code-by-removing-unneeded-indentation-3g9i

```suggestion
    if (!m_useBias)
    {
        return true;
    }

    if (!m_stateVariableHandler.getVariable(m_biasVariableName).isValid())
    {
        log()->error("{} The variable handler does not contain the expected state with name `{}`.", errorPrefix, m_biasVariableName);
         return false;
    }
    return true;
```